packages/@azure/servicebus/data-plane/test/sessionsTests.spec.ts

```diff
@@ -0,0 +1,294 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import chai from "chai";
+const should = chai.should();
+import chaiAsPromised from "chai-as-promised";
+import dotenv from "dotenv";
+dotenv.config();
+chai.use(chaiAsPromised);
+import {
+  Namespace,
+  QueueClient,
+  TopicClient,
+  SubscriptionClient,
+  MessageSession,
+  delay,
+  ServiceBusMessage
+} from "../lib";
+
+import {
+  testMessagesWithSessions,
+  testMessagesWithDifferentSessionIds,
+  getSenderClient,
+  getSessionClient,
+  ClientType,
+  testSessionId
+} from "./testUtils";
+
+async function testPeekMsgsLength(
+  client: QueueClient | SubscriptionClient | MessageSession,
+  expectedPeekLength: number
+): Promise<void> {
+  const peekedMsgs = await client.peek(expectedPeekLength + 1);
+  should.equal(
+    peekedMsgs.length,
+    expectedPeekLength,
+    "Unexpected number of msgs found when peeking"
+  );
+}
+
+let ns: Namespace;
+
+let senderClient: QueueClient | TopicClient;
+let sessionClient: QueueClient | SubscriptionClient;
+
+let unexpectedError: Error | undefined;
+
+function unExpectedErrorHandler(err: Error): void {
+  if (err) {
+    unexpectedError = err;
+  }
+}
+
+async function beforeEachTest(senderType: ClientType, sessionType: ClientType): Promise<void> {
+  // The tests in this file expect the env variables to contain the connection string and
+  // the names of empty queue/topic/subscription that are to be tested
+
+  if (!process.env.SERVICEBUS_CONNECTION_STRING) {
+    throw new Error(
+      "Define SERVICEBUS_CONNECTION_STRING in your environment before running integration tests."
+    );
+  }
+
+  ns = Namespace.createFromConnectionString(process.env.SERVICEBUS_CONNECTION_STRING);
+
+  senderClient = getSenderClient(ns, senderType);
+  sessionClient = getSessionClient(ns, sessionType);
+}
+
+async function afterEachTest(): Promise<void> {
+  await ns.close();
+}
+
+describe("Accept a session by passing non-existing sessionId receives no messages", function(): void {
+  afterEach(async () => {
+    await afterEachTest();
+  });
+
+  async function test_batching(): Promise<void> {
+    await senderClient.send(testMessagesWithSessions[0]);
+
+    let receiverClient = await sessionClient.acceptSession({ sessionId: "non" + testSessionId });
+    let msgs = await receiverClient.receiveBatch(1, 10);
+    should.equal(msgs.length, 0);
+
+    await receiverClient.close();
+    receiverClient = await sessionClient.acceptSession();
+    await testPeekMsgsLength(receiverClient, 1);
```

Comment:
This peek is not required, instead we can check that the `msgs` array in the next line has only 1 element

Reply:
Same for `test_streaming`